src/MongoDB/ServerDescription.c

```diff
@@ -0,0 +1,347 @@
+static PHP_METHOD(ServerDescription, isPassive)
+{
+	php_phongo_serverdescription_t* intern;
+	bson_iter_t                     iter;
+
+	intern = Z_SERVERDESCRIPTION_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	RETVAL_BOOL(bson_iter_init_find_case(&iter, mongoc_server_description_hello_response(intern->server_description), "passive") && bson_iter_as_bool(&iter));
+} /* }}} */
+
+/* {{{ proto boolean MongoDB\Driver\ServerDescription::isPrimary()
+   Returns whether the server is a primary member of a replica set */
+static PHP_METHOD(ServerDescription, isPrimary)
+{
+	php_phongo_serverdescription_t* intern;
+
+	intern = Z_SERVERDESCRIPTION_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	RETVAL_BOOL(!strcmp(mongoc_server_description_type(intern->server_description), php_phongo_server_description_type_map[PHONGO_SERVER_RS_PRIMARY].name));
+} /* }}} */
+
+/* {{{ proto boolean MongoDB\Driver\ServerDescription::isSecondary()
+   Returns whether the server is a secondary member of a replica set */
+static PHP_METHOD(ServerDescription, isSecondary)
+{
+	php_phongo_serverdescription_t* intern;
+
+	intern = Z_SERVERDESCRIPTION_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	RETVAL_BOOL(!strcmp(mongoc_server_description_type(intern->server_description), php_phongo_server_description_type_map[PHONGO_SERVER_RS_SECONDARY].name));
+} /* }}} */
+
+/* {{{ MongoDB\Driver\ServerDescription function entries */
+ZEND_BEGIN_ARG_INFO_EX(ai_ServerDescription_void, 0, 0, 0)
+ZEND_END_ARG_INFO()
+
+static zend_function_entry php_phongo_serverdescription_me[] = {
+	/* clang-format off */
+	PHP_ME(ServerDescription, getHelloResponse, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getHost, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getId, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getLastUpdateTime, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getPort, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getRoundTripTime, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getTags, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getType, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, isArbiter, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, isHidden, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, isPassive, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, isPrimary, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, isSecondary, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
```

Comment:
I realize these were lifted from the Server class, but since this is a new API I'd prefer to keep it minimal until someone actually requests these. Let's limit it to the relevant methods on [mongoc_server_description_t](http://mongoc.org/libmongoc/current/mongoc_server_description_t.html):

 * `getHelloResponse`
 * `getHost`
 * `getPort`
 * `getLastUpdateTime`
 * `getRoundTripTime`
 * `getType`

We can omit `getId` since that's just the internal identifier libmongoc uses and not really relevant to our users.

As far as reporting this info in debug output, we can use the following fields:

 * `hello_response`
 * `host`
 * `port`
 * `last_update_time`
 * `round_trip_time`
 * `type`

Since we've yet to drop support for 32-bit, I think you can lift logic we use elsewhere to conditionally report `last_update_time` and `round_trip_time` as strings if needed (using existing macros). I'm not too worried about the methods themselves -- those can always return a `zend_long` type.